### PR TITLE
Remove alphanumeric regex on involvement

### LINKED ISSUE #224: [PE-D] UG does not mention that the [INVOLVEMENT] parameter must be alphanumeric in filterStudent
I presume that the filter function is only meant to take in alphanumeric characters for both parameters. The error message also does not mention why the command is invalid.

Command run: filterStudent class monitor, prefect

-------------
Labels: `type.DocumentationBug` `severity.Low`
original: neojxuan/ped#6

### PR DESCRIPTION
**Pull request by cliveong**
Apply slap on nested `processInput `

Rename methods to better fit what they are doing

Add information on message usage for filtercommands

Fixed **key issue 228**, and other minor typos at the same time

### CODE REVIEW

File: src/main/java/seedu/address/logic/parser/student/FilterStudentCommandParser.java [hunk 1]
```diff
@@ -41,41 +41,36 @@ public FilterStudentCommand parse(String args) throws ParseException {
...
+        }
+        return listToReturn;
+    }
+
+    private boolean processEachWord(String[] keywords, boolean isTagsOnwards, List<String> listToReturn, int i)
+            throws ParseException {
+        if (!isTagsOnwards) {
+            if (keywords[i].startsWith("t/") && isValidFirstTag(keywords[i])) {
+                isTagsOnwards = true;
+                listToReturn.add(keywords[i]);
```
Comment (junghon3709) [inline src/main/java/seedu/address/logic/parser/student/FilterStudentCommandParser.java:54]: Alright, I wish the code could be neater as well, but it seems like there's nothing much we can do about it at this late stage. Could you perhaps add in some javadocs before I merge it? Ie. maybe give a toy example of how a filterStudent command gets passed down, and how parameters are parsed and passed around.

**Review by junghon3709 — COMMENTED** [review]
Alright, just add some javadocs to make it clearer.

File: src/main/java/seedu/address/logic/parser/student/FilterStudentCommandParser.java (hunk 1 above)
Comment (cliveong) [reply src/main/java/seedu/address/logic/parser/student/FilterStudentCommandParser.java:62]: done

File: src/main/java/seedu/address/logic/parser/student/FilterStudentCommandParser.java [hunk 2]
```diff
@@ -41,41 +41,36 @@ public FilterStudentCommand parse(String args) throws ParseException {
...
-                if (!keywords[i].substring(2).matches(TAG_REGEX)) {
-                    throw new ParseException(
-                            String.format(MESSAGE_INVALID_COMMAND_FORMAT, FilterStudentCommand.MESSAGE_USAGE));
-                }
+            isTagsOnwards = processEachWord(keywords, isTagsOnwards, listToReturn, i);
+        }
+        return listToReturn;
+    }
+
+    private boolean processEachWord(String[] keywords, boolean isTagsOnwards, List<String> listToReturn, int i)
```
[the remark this answers is not in the document]
Comment (cliveong) [reply src/main/java/seedu/address/logic/parser/student/FilterStudentCommandParser.java]: done

**Pull request merged**